src/test/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/RunTestFolderStepTest.groovy

```diff
@@ -0,0 +1,189 @@
+
+        then:
+            thrown(AbortException)
+
+        where:
+            isAbsolute  | exists
+            false       | true
+            true        | false
+
+    }
+
+    def "scanPackages returns list of package files"() {
+        given:
+            def step = new RunTestFolderStep(baseFolder)
+            GroovyMock(TestPackageScanner, global: true)
+            def packageScanner = Mock(TestPackageScanner)
+            new TestPackageScanner(_, _, _) >> packageScanner
+            packageScanner.scanTestFiles() >> expectedPackages
+
+        when:
+            def result = step.scanPackages("", context, scanMode, false)
+
+        then:
+            result == expectedPackages
+            if (expectedPackages.isEmpty() && scanMode != RunTestFolderStep.ScanMode.PROJECTS_ONLY ){
+                1 * context.get(TaskListener.class).logger.println('No packages found!')
+            }
+            else if (scanMode != RunTestFolderStep.ScanMode.PROJECTS_ONLY){
+                1 * context.get(TaskListener.class).logger.println("Found ${expectedPackages.size()} package(s)")
+            }
+
+        where:
+            scanMode                                    | expectedPackages
+            RunTestFolderStep.ScanMode.PROJECTS_ONLY    | []
+            RunTestFolderStep.ScanMode.PACKAGES_ONLY    | []
+            RunTestFolderStep.ScanMode.PACKAGES_ONLY    | ["package.pkg","package2.pkg"]
+    }
+
+    def "scanProjects returns list of project files"() {
+        given:
+            def step = new RunTestFolderStep(baseFolder)
+            GroovyMock(TestProjectScanner, global: true)
+            def projectScanner = Mock(TestProjectScanner)
+            new TestProjectScanner(_, _, _) >> projectScanner
+            projectScanner.scanTestFiles() >> expectedProjects
+
+        when:
+            def result = step.scanProjects("", context, scanMode, false)
+
+        then:
+            result == expectedProjects
+            if (expectedProjects.isEmpty() && scanMode != RunTestFolderStep.ScanMode.PACKAGES_ONLY ){
+                1 * context.get(TaskListener.class).logger.println('No projects found!')
+            }
+            else if (scanMode != RunTestFolderStep.ScanMode.PACKAGES_ONLY){
+                1 * context.get(TaskListener.class).logger.println("Found ${expectedProjects.size()} project(s)")
+            }
+
+        where:
+            scanMode                                    | expectedProjects
+            RunTestFolderStep.ScanMode.PACKAGES_ONLY    | []
+            RunTestFolderStep.ScanMode.PROJECTS_ONLY    | []
+            RunTestFolderStep.ScanMode.PROJECTS_ONLY    | ["project1.prj","project2.prj"]
+    }
+
+    def "run method executes tests based on failFast setting"() {
+        given:
+            GroovySpy(RunTestFolderStep, global: true)
+            RunTestFolderStep.scanPackages(*_) >> ["package1.pkg", "package2.pkg"]
+            RunTestFolderStep.scanProjects(*_) >> []
```

Comment:
This only tests the fail fast for package executions, but not for project executions